addon/adapters/osf-adapter.js

```diff
@@ -10,14 +11,82 @@ export default DS.JSONAPIAdapter.extend(DataAdapterMixin, {
             url += '/';
         }
         return url;
+    },
+    /**
+     * Build the request payload for a relationship create/update. We're
+     * using the meta hash of the relationship field to pass an optional
+     * custom serialization method. This bypasses the normal serialization
+     * flow, but is necessary to cooperate with the OSF APIv2.
+     *
+     * @method _buildRelationshipPayload
+     * @param {DS.Store} store
+     * @param {DS.Snapshot} snapshot
+     * @param {String} relationship the relationship to build a payload for
+     * @return {Object} the serialized relationship
+     **/
+    _relationshipPayload(store, snapshot, relationship) {
+        var relationMeta = snapshot.record[relationship].meta();
+        var relationType = relationMeta.type;
+        var serialized;
+        if (relationMeta.options.serializer) {
+            serialized = relationMeta.options.serializer(snapshot.record);
+        } else {
+            var serializer = store.serializerFor(relationType);
+            var toBeSent = snapshot.record.get(relationship).filter(record => record.id === null);
+            serialized = serializer.serialize(toBeSent);
+        }
+        return serialized;
+    },
+    /**
+     * Construct a URL for a relationship create/update/delete. Has the same
+     * signature as buildURL, with the addition of a 'relationship' param
+     *
+     * @method _buildRelationshipURL
+     * @param {String} relationship the relationship to build a url for
+     * @return {String} a URL
+     **/
+    _buildRelationshipURL() {
+        var [, , snapshot, , , relationship] = arguments;
+        var links = relationship ? snapshot.record.get(
+            `relationshipLinks.${Ember.String.underscore(relationship)}.links`
+        ) : false;
+        if (links) {
+            return links.self ? links.self.href : links.related.href;
+        } else {
+            return this.buildURL(...arguments);
+        }
+    },
+    updateRecord(store, type, snapshot, _, query) {
+        var promises = [];
+
+        var dirtyRelationships = snapshot.record.get('dirtyRelationships');
+        if (dirtyRelationships.length) {
+            promises = promises.concat(dirtyRelationships.map(relationship => {
+                var url = this._buildRelationshipURL(type.modelName, snapshot.id, snapshot, 'updateRecord', query, relationship);
+                var requestType = snapshot.record[relationship].meta().options.updateRequestType;
+                return this.ajax(url, requestType || 'PATCH', {
+                    data: this._relationshipPayload(store, snapshot, relationship)
+                }).then(() => snapshot.record.clearDirtyRelationship(relationship));
+            }));
+        }
+        if (Object.keys(snapshot.record.changedAttributes()).length) {
+            promises.push(this._super(...arguments));
+        }
+        return Ember.RSVP.Promise.all(promises).then(values => {
```

Comment:
Docs on this method aren't great- but it appears that this will be rejected at the first failure. What happens if one save succeeds, but others fail?

This might be relevant and useful:
http://emberjs.com/api/classes/RSVP.html#method_allSettled